sycl/source/detail/device_impl.cpp

```diff
@@ -301,6 +303,22 @@ std::shared_ptr<device_impl> device_impl::getHostDeviceImpl() {
   return HostImpl;
 }
 
+bool device_impl::isAssertFailSupported() const {
+  // assert is sort of natively supported by host
+  if (MIsHostDevice)
+    return true;
+
+  const plugin &Plugin = getPlugin();
+
+  // Return false for CUDA even though, it has native support for assert.
+  // Change to return true when devicelib issue is fixed.
+  // See https://github.com/intel/llvm/issues/3385
+  if (Plugin.getBackend() == backend::cuda)
+    return false;
+
+  return has_extension("cl_intel_devicelib_cassert");
```

Comment:
Should check for `PI_INTEL_DEVICELIB_CASSERT`, not `cl_intel_...`

Reply:
Fixed.

Reply:
@smaslov-intel , could you, please, review PI changes once more? I've updated PI API so that it allows backend specific extension naming for this particular case. I think this can be found useful in future also.